jumanji/specs_test.py

```diff
@@ -15,202 +15,162 @@
 # Adapted from dm_env.specs_test
 # ============================================================================
 import pickle
-from typing import Any, Iterable, NamedTuple, Sequence, Union
+from collections import namedtuple
+from typing import Any, NamedTuple, Sequence, Union
 
 import chex
 import dm_env.specs
 import gym.spaces
 import jax.numpy as jnp
-import jax.tree_util
 import numpy as np
 import pytest
 from chex import assert_trees_all_equal
 
 from jumanji import specs
+from jumanji.testing.pytrees import assert_tree_with_leaves_of_type
 
 
-class ThirdClass(NamedTuple):
+class SinglyNested(NamedTuple):
```

Comment:
I would probably go with `SingleNestedNamedTuple` I feel it makes it clearer on what it is